### ISSUE TITLE
Large shared folder use case question

### ISSUE THREAD

**Issue by jmrichardson** [jbarlow83/OCRmyPDF#256]
Hoping someone could offer a best practice or way to solution this use case:  I have a 6M+ file shared folder (NFS) that I would like to use OCRmyPDF for all PDFs stored.  I do not want to run a batch/directory job that has to check all files and process as needed (too time consuming).  I can't use inotify tools either because it's a windows based NFS share to process files as added.  Is there a way to store a database of files already processed (as an index) and skip over those files on next nightly run?  Something similar?  Thanks in advance.

**Comment by jbarlow83** [comment]
There's nothing internal to ocrmypdf that does that.

I wouldn't trust anyone who suggests a "best practice" that involves the words "Windows-based NFS share". (I guess this is to work around a Windows file server that holds the files while a Linux box is running OCRmyPDF? OCRmyPDF is probably not far from running natively Windows support, if you wanted to try it out in a PR. The main reason I've avoided it is the difficulty of handling external-to-Python dependencies like Tesseract and Ghostscript.)

Anyway, I'd set up a folder tree that represents all files that need OCR. When a file is uploaded put it in both its target location and a mirror location in the OCR queue (assuming you want users to have immediate access to the non-OCR version). The trees should contain the same folder trees:

```
files/
  invoices/
    2018/
      01.pdf
      02.pdf
  receipts/
  bills/
ocrqueue/
  invoices/
    2018/
      02.pdf
  receipts/
  bills/
```

Every night, scan ocrqueue. On success, overwrite the version in files/ location and delete the version in the OCR queue. Arrange for ocrqueue to be hidden from users. This should work in theory, although you have NFS to consider. 

I've also used Git LFS for this sort of thing. For every untracked file, run OCR and commit. But don't do that over a networked file system.
